ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -161,8 +153,9 @@ void handleResponse(ResponseInfo responseInfo) {
     GetOperation getOperation = correlationIdToGetOperation.remove(getRequest.getCorrelationId());
     if (getOperations.contains(getOperation)) {
       getOperation.handleResponse(responseInfo);
-      if (getOperation.isOperationComplete() && getOperations.remove(getOperation)) {
-        onComplete(getOperation);
+      if (getOperation.isOperationComplete()) {
+        getOperations.remove(getOperation);
+        routerMetrics.operationDequeuingRate.mark();
```

Comment:
yeah, not that it is a big deal if the metric gets counted during close, but since this can be easily handled, will make the change. Introduced a method to remove from `getOperations`. Anytime we remove, we will need to update this (and the get rate metric which will come in later).